New Poker Cheat Sheet

jorge305 opened the pull request:
https://duck.co/ia/view/hold_em_starting_hands_cheat_sheet

daxtheduck commented:
@jorge305 Hey!
        
Thanks for taking the time to contribute! We really appreciate it.

We work closely with every contributor to make Instant Answers the best they can be, so we appreciate your patience as we look over your code. From here, the process usually goes like this:

1. Pull Request is reviewed by the DuckDuckGo staff and community
2. Staff and community will leave feedback with any necessary updates to the function or design.
3. Once you've made any necessary corrections, then your Instant Answer will be merged and deployed live on DuckDuckGo!

If you have any questions along the way, feel free to ask them here. Our staff and community are also available on Slack to answer any questions you may have. If you'd like to join us there please send an email to <quackslack@duckduckgo.com>.

>More Info: https://duck.co/duckduckhack/submission_and_review

Thanks!

daxtheduck commented:
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

moollaza commented:
@jorge305 thanks! I think most of this is actually covered by our existing Poker Hands Cheat Sheet?

https://duckduckgo.com/?q=poker+hands+cheat+sheet&ia=cheatsheet

Can you take a look and compare? Do you have any suggested updates for that cheat sheet perhaps?

jorge305 commented:
After taking a look, I think they might be different and worth two cheat sheets.

https://duckduckgo.com/?q=poker+hands+cheat+sheet&ia=cheatsheet

has the rankings of the final hands in poker and the odds of being dealt those hands in 5 card poker.

The cheat sheet I submitted has the odds of starting hands being dealt in the hold'em variant of poker which is different from 5 card poker. 

Let me know your thoughts please.

moollaza commented:
> The cheat sheet I submitted has the odds of starting hands being dealt in the hold'em variant of poker which is different from 5 card poker.

Thanks for clarifying -- clearly I'm not much of a poker player ;)

moollaza commented:
@jorge305 I've set this to deploy to Beta. I wanted to point out though that none of your example queries are actually going to trigger this Cheat Sheet because of the way cheat sheet triggers work, which for now, requires that the file name (or an `alias`) is in the query, along with one of our trigger words.

It may be Goodie to also build a Goodie that gives the specific percentages for the kinds of queries you  have there.

I'm going to update the example query to one that will actually trigger this Cheat Sheet though so the screenshot works as expeted :+1:

moollaza commented on share/goodie/cheat_sheets/json/hold-em-starting-hands.json:
```diff
@@ -0,0 +1,53 @@
+    },
+
+    "aliases": [
+        "first alias", "odds of being dealt aa", "odds of being dealt aces", "odds of being dealt a pair", "odds of being dealt suited connectors", "hold'em starting hands", "odds of being dealt suited cards", "odds of being dealt pocket kings", "odds of being dealt pocket aces", "no limit starting hands", "texas holdem starting hands" 
```
let's remove `first alias` and the `odds of ...` aliases

moollaza replied:
you may want to add `hold'em` and `texas hold'em` as aliases

jorge305 commented:
@moollaza Thanks for your help.

I fixed up the aliases as you suggested and updated the IA page with example queries that will work.

I made the changes and push them to my branch but do I need to do anything to the pull request I created? (sorry rookie here)

Thanks again for your help

mohan08p commented:
@jorge305 : remove capitals from aliases... [ never used caps in alias section ! ]

jorge305 commented:
@mohan08p sorry about the caps in the aliases and that it took me so long to fix it.

Please let me know if I can do anything else

daxtheduck commented:
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=Holdem Starting Hands Cheat Sheet)
                ![](https://ia-screenshots.s3.amazonaws.com/hold_em_starting_hands_cheat_sheet_index.png)
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/hold_em_starting_hands_cheat_sheet_mobile.png)

mohan08p commented:
@jorge305 : No problem, you are good to go. :+1:

edgesince84 commented:
@jorge305,  thanks again for creating the Poker Cheat Sheet! You’ve done a really good job here and we'd like you to help our community tackle the next big challenge - our mission to become the best search engine for programmers: https://duck.co/blog/post/297/help-for-programmers.

When you're ready to move on, please: 

- Create a new Instant Answer to increase our Programming IA coverage

OR

- Test live programming Instant Answers (or those in development) for bugs and suggest improvements 

Detailed instructions explaining how you can help are available at the top of this [page](https://github.com/duckduckgo/duckduckgo/wiki/Programming-IA-Coverage).

Thanks again!

moollaza commented:
@jorge305 Thanks! This LGTM :+1:

daxtheduck commented:
@jorge305 this is now merged! It should be live on DuckDuckGo within one week (or during the next release) and we'll be sure to notify you when it happens.

After your Instant Answer goes live for the World to use, you will be able to see [traffic information](http://docs.duckduckhack.com/maintaining/tools.html#monitor-your-instant-answer-page) about your IA as well as receive feedback directly from users. 
                
Please refer to the [maintenance guide and docs](http://docs.duckduckhack.com/maintaining/guidelines.html) to keep up with potential changes to your Instant Answer.
                
Thank you again for your help!
